internal/scorecard/alpha/scorecard.go

```diff
@@ -117,29 +133,31 @@ func ConfigDocLink() string {
 		version.Version)
 }
 
-// waitForTestsToComplete waits for a fixed amount of time while
-// checking for test pods to complete
-func (o Scorecard) waitForTestsToComplete(tests []Test) (err error) {
+// waitForTestToComplete waits for a fixed amount of time while
+// checking for a test pod to complete
+func (o Scorecard) waitForTestToComplete(ctx context.Context, p *v1.Pod) (err error) {
 	waitTimeInSeconds := int(o.WaitTime.Seconds())
 	for elapsedSeconds := 0; elapsedSeconds < waitTimeInSeconds; elapsedSeconds++ {
-		allPodsCompleted := true
-		for _, test := range tests {
-			p := test.TestPod
-			var tmp *v1.Pod
-			tmp, err = o.Client.CoreV1().Pods(p.Namespace).Get(context.TODO(), p.Name, metav1.GetOptions{})
-			if err != nil {
-				return fmt.Errorf("error getting pod %s %w", p.Name, err)
-			}
-			if tmp.Status.Phase != v1.PodSucceeded {
-				allPodsCompleted = false
-			}
-
+		var tmp *v1.Pod
+		tmp, err = o.Client.CoreV1().Pods(p.Namespace).Get(ctx, p.Name, metav1.GetOptions{})
+		if err != nil {
+			return fmt.Errorf("error getting pod %s %w", p.Name, err)
 		}
-		if allPodsCompleted {
+		if tmp.Status.Phase == v1.PodSucceeded {
 			return nil
 		}
+
 		time.Sleep(1 * time.Second)
 	}
 	return fmt.Errorf("error - wait time of %d seconds has been exceeded", o.WaitTime)
```

Comment:
This function should be rewritten to use [`wait.UntilWithContext()`](https://github.com/kubernetes/apimachinery/blob/15d95c0b2af3f4fcf46dce24105e5fbb9379af5a/pkg/util/wait/wait.go#L98)

After we've created the context from `o.WaitTime`, the context should be what drives timeout and cancellation for the whole run.